UI: Dex integration

carlito-scality opened the pull request:
**Component**: UI 

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 
- Integrate Dex login using redux-oidc library

**Summary**:

**Acceptance criteria**: 
- The MetalK8s UI should directly use dex to authenticate a user using OAuth2 flow.
- Once logged in, the reste should works like before without regressions

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: https://github.com/scality/metalk8s/issues/2015

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

bert-e commented:
# Conflict




There is a conflict between your branch `feature/dex-integration-in-ui` and the
destination branch `development/2.4`.

Please resolve the conflict on **the feature branch** (`feature/dex-integration-in-ui`).

```bash
 $ git fetch
 $ git checkout origin/feature/dex-integration-in-ui
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/dex-integration-in-ui
```

carlito-scality committed:
UI: integrate Dex using redux-oidc
- re-implement the login workflow using redux-oidc library
- update the way to authenticate to Salt
- remove old Login page and image

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality committed:
UI: refresh page management
- catch the redux-oidc USER_FOUND action, to update ApiK8s services and authenticate to salt
- add isUserLoaded state to check if the user is loaded

carlito-scality committed:
update local config.json with url_oidc_provider and url_redirect

gdemonet commented:
/after_pull_request=2063

bert-e commented:
# Waiting for other pull request(s)



The current pull request is locked by the **after_pull_request** option.

In order for me to merge this pull request, run the following actions first:


:arrow_right: Merge the `OPEN` pull request:

* pull request #2063





Alternatively, delete all the **after_pull_request** comments from this pull request.





*The following options are set:* **after_pull_request**

gdemonet commented:
@carlito-scality you'll need to rebase on `development/2.5` when #2063 gets merged.

gdemonet commented:
/after_pull_request=2025

bert-e commented:
# Waiting for other pull request(s)



The current pull request is locked by the **after_pull_request** option.

In order for me to merge this pull request, run the following actions first:


:arrow_right: Merge the `OPEN` pull requests:

* pull request #2025

* pull request #2063





Alternatively, delete all the **after_pull_request** comments from this pull request.





*The following options are set:* **after_pull_request**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **after_pull_request**

carlito-scality committed:
UI: Update config.json configMap to add url_oidc_provider and url_redirect urls
- url_oidc_provider allow to fetch /oidc/.well-known/openid-configuration file
- url_redirect should contain full URL since it should be regirstered as it is in the DEX config and it is used for redirection once the login in DEX login form is successful.
Ref: #2015

carlito-scality committed:
UI: improve the Login Redirecting page
- After the login in Dex Form is successful, we go back to the CallbackPage containing a Loader with the text
Ref: #2015

carlito-scality committed:
UI: update cypress tests to support DEX
- update cypress.json with new default login
- add return null to PrivateRoute since it should always return something. Otherswise, we got an JS error.
Ref: #2015

carlito-scality committed:
UI: update tests relative to login
Ref: #2015

carlito-scality committed:
UI: Display only username once logged in
- Display only username instead of username + email
- update cypress tests
Ref: #2015

TeddyAndrieux commented on salt/metalk8s/addons/ui/deployed/ui.sls:
```diff
@@ -1,6 +1,12 @@
 - .namespace
 
+{%- set kubeconfig = "/etc/kubernetes/admin.conf" %}
+{%- set context = "kubernetes-admin@kubernetes" %}
```
This is no longer needed, please remove it

ChengYanJin reviewed (dismissed):
LGTM! Great job!!!
We will need to handle share user data between tabs in order to eliminate unnecessary authentication.

carlito-scality committed:
UI: remove useless code
- it is introduced during last rebasing

TeddyAndrieux committed:
salt: Fix `ingress_control_plane` for `salt-run`

When we try to retrieve informations from the mine from salt-master we
cannot use module `mine.get` we need to use `saltutil.runner` with
`mine.get`, this approach works for salt-master but not for
salt-minion not on the same machine as the salt-master

carlito-scality committed:
UI: increase the limit the maxAssetSize
-We add redux-oidc for dex login which increase the bundle size.

TeddyAndrieux reviewed (approved):
backend part LGTM

carlito-scality commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, after_pull_request**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye carlito-scality.
